cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/ApiCredentials.java

```diff
@@ -0,0 +1,337 @@
+ * authenticating to Cruise Control's REST API.
+ */
+public class ApiCredentials {
+    // Regex to match an entry in Jetty's HashLoginService's file format: username: password, rolename
+    private static final Pattern HASH_LOGIN_SERVICE_PATTERN = Pattern.compile("^[\\w-]+\\s*:\\s*\\w+\\s*,\\s*\\w+\\s*$");
+    private static final List<String> FORBIDDEN_USERNAMES = Arrays.asList(
+            HEALTHCHECK_USERNAME,
+            REBALANCE_OPERATOR_USERNAME,
+            TOPIC_OPERATOR_USERNAME
+    );
+
+    private final String userManagedApiSecretName;
+    private final String userManagedApiSecretKey;
+    private final boolean userManagedApiUsersEnabled;
+    private final String namespace;
+    private final String cluster;
+    private final Labels labels;
+    private final CruiseControlSpec specSection;
+    private final OwnerReference ownerReference;
+
+    /**
+     * Constructs the Api Credentials Model for managing API users for Cruise Control API.
+     *
+     * @param namespace         Namespace of the cluster
+     * @param cluster           Name of the cluster to which this component belongs
+     * @param labels            Labels for Cruise Control instance
+     * @param ownerReference    Owner reference for Cruise Control instance
+     * @param specSection       Custom resource section configuring Cruise Control API users
+     */
+    public ApiCredentials(String namespace, String cluster, Labels labels, OwnerReference ownerReference, CruiseControlSpec specSection) {
+        this.namespace = namespace;
+        this.cluster = cluster;
+        this.labels = labels;
+        this.ownerReference = ownerReference;
+        this.specSection = specSection;
+
+        userManagedApiUsersEnabled = validateApiUsersConfig(specSection);
+        if (userManagedApiUsersEnabled) {
+            userManagedApiSecretName = specSection.getApiUsers().getValueFrom().getSecretKeyRef().getName();
+            userManagedApiSecretKey = specSection.getApiUsers().getValueFrom().getSecretKeyRef().getKey();
+        } else {
+            userManagedApiSecretName = null;
+            userManagedApiSecretKey = null;
+        }
+    }
+
+    /**
+     * @return  Returns true if user-managed API user config is enabled
+     */
+    public boolean userManagedApiUsersEnabled() {
+        return userManagedApiUsersEnabled;
+    }
+
+    /**
+     * @return  Returns user-managed API credentials secret name
+     */
+    public String getUserManagedApiSecretName() {
+        return userManagedApiSecretName;
+    }
+
+    /**
+     * @return  Returns user-managed API credentials secret key
+     */
+    /* test */ String getUserManagedApiSecretKey() {
+        return userManagedApiSecretKey;
+    }
+
+    /**
+     * Checks if Cruise Control spec has valid ApiUsers config.
+     *
+     * @param ccSpec The Cruise Control spec to check.
+     */
+    private static boolean validateApiUsersConfig(CruiseControlSpec ccSpec)  {
```

Comment:
If it validates, it should not handle whether it is set or not. When the user does not use the API users, it is a valid configuration.

Reply:
Reverted